bubble_sort.rb

```diff
@@ -0,0 +1,43 @@
+arr = [4, 3, 78, 2, 0, 2]
+
+def bubble_sort_by(my_arr)
+  yield my_arr
+end
+bubble_sort_by(arr) do |x, y|
+  arr.length.times do
+    i = 0
+    while i < arr.length - 1
+      x = arr[i]
+      y = arr[i + 1]
+      sorted = x <=> y
+      arr.slice!(i, 2)
+      if sorted == -1
+        arr.insert(i, x, y)
+      else
+        arr.insert(i, y, x)
+      end
+      i += 1
+    end
+  end
+  puts arr.to_s
+end
```

Comment:
- [ ] sorts an array by accepting a block
`Below is a sample test case that should work if `*bubble_sort_by*` is well implemented. Use it as a reference to make your method work as expected.`
```ruby
unsorted = (1..10).to_a.reverse!
bubble_sort_by(unsorted) { |left, right| left.length - right.length } == %w[hi hey hello] # should return true
```